### ISSUE TITLE
add Qwen2-VL/Qwen2.5-VL

### ISSUE THREAD

**Issue by FelisDwan** [ollama/ollama#6564]
SOTA light weight vision model
[https://github.com/QwenLM/Qwen2-VL](https://github.com/QwenLM/Qwen2-VL)

llama.cpp issue [#9246](https://github.com/ggerganov/llama.cpp/issues/9246)

**Comment by XDesktopSoft** [comment]
+1

**Comment by SP4595** [comment]
support +1

**Comment by Halfknow** [comment]
hope to support

**Comment by BBBOND** [comment]
+1

**Comment by lojik-ng** [comment]
+1

**Comment by elyzionz** [comment]
+1

**Comment by zyfewq** [comment]
+1

**Comment by pmddds** [comment]
+1

**Comment by zwqjoy** [comment]
hope to support

**Comment by stephonye** [comment]
** I am really looking forward to this model being used in Ollama**

**Comment by kalkishnu** [comment]
yes pls.

**Comment by dannykhoo** [comment]
support +1

**Comment by AlanceLau** [comment]
support +1

**Comment by AlanceLau** [comment]
support +1

**Comment by zhaoqi571436204** [comment]
+1

**Comment by Alex-gy** [comment]
+1 Please

**Comment by arkerwu** [comment]
+1

**Comment by fidojones** [comment]
+1

**Comment by enryteam** [comment]
+1

**Comment by cwbjyy** [comment]
+1

**Comment by a1558878** [comment]
hope to support

**Comment by ddpasa** [comment]
Benchmarks look very strong, please support this model.

**Comment by shivanshkaushikk** [comment]
+1

**Comment by Erionis** [comment]
+1

**Comment by Milor123** [comment]
+1 ❤️

**Comment by yblis** [comment]
+1

**Comment by alivekevin** [comment]
+1

**Comment by Reknine** [comment]
+1

**Comment by 0dram** [comment]
+1

**Comment by trackcan** [comment]
+1

**Comment by Axenide** [comment]
+1

**Comment by PredyDaddy** [comment]
+1

**Comment by lyfuci** [comment]
+1

**Comment by haddyadnan** [comment]
+1

**Comment by sebastianhorwege** [comment]
+1

**Comment by Tarakzai** [comment]
+1

**Comment by Osenzwerg** [comment]
+1

**Comment by bunnyfu** [comment]
Here to support :)

**Comment by isriam** [comment]
+1 this model was able to read recipes that other models failed at and i was using openai to ocr with.

**Comment by tomyjany** [comment]
here to support :) soon there will be the 72 version

**Comment by devsanbid** [comment]
+1

**Comment by selimcavas** [comment]
+1

**Comment by shreyvish5678** [comment]
when will this be added? I've been looking to use it

**Comment by khwengXU** [comment]
+1

**Comment by huaji23** [comment]
+1

**Comment by ankykong** [comment]
+1

**Comment by iwannabewater** [comment]
hope to add qwen2-vl soon, thank you all guys!

**Comment by jasondalycan** [comment]
+1

**Comment by wangyuc** [comment]
+1

**Comment by FickleLife** [comment]
+1 !!!

**Comment by zhaoyang0618** [comment]
+1!!!

**Comment by erinsrc** [comment]
+1 Please!

**Comment by zhaolj** [comment]
+1 !!!

**Comment by utopeadia** [comment]
+1

**Comment by Jakolo121** [comment]
+1

**Comment by junqiang1992** [comment]
+1 Please!

**Comment by regoo** [comment]
+++！

**Comment by chaoqunxie** [comment]
++1

**Comment by misteral** [comment]
+1

**Comment by gokayfem** [comment]
++++

**Comment by bughoho** [comment]
+++++

**Comment by supperdsj** [comment]
+++++++++

**Comment by utopeadia** [comment]
Hey guys, let's stop the +1 or +++ activities, this is a "mail disaster" for each subscriber, just liking the existing replies is enough

**Comment by dr-ijaz** [comment]
+++1

**Comment by yak0** [comment]
++++1

**Comment by netixc** [comment]
+1

**Comment by eignerfr** [comment]
+1

**Comment by damianpdr** [comment]
+1

**Comment by love01211** [comment]
+1

**Comment by ammoti** [comment]
one more +1

**Comment by Rakhsan** [comment]
+1

**Comment by vardaan123** [comment]
+1

**Comment by fovendor** [comment]
+1

**Comment by xmlvrl** [comment]
hope to support

**Comment by cczw2010** [comment]
+1 please

**Comment by gmarkantonatos** [comment]
+1 The model is great!

**Comment by yossifibrahem** [comment]
Add my voice with you

**Comment by jochne** [comment]
+1
plz

**Comment by LuWei6896** [comment]
++++++++1

**Comment by Linnore** [comment]
+1

**Comment by Rakhsan** [comment]
+10^10 man pls man ollama main dev pls

**Comment by Syazvinski** [comment]
this is dumb, its been almost 5 months. 

Add it

+1

**Comment by yyqqing** [comment]
+1

**Comment by ALLinLLM** [comment]
Qwen2-VL work better than llama3.2-vision 11B for pictures contain Chinese Character, so just add it plz

**Comment by Syazvinski** [comment]
Ollama is based on Llama.cpp and guff models, qwen VL is not compatible with llama.cpp as it uses a different archetechure because of its vision. it wont be added anytime soon likley.

**Comment by ddpasa** [comment]
> Ollama is based on Llama.cpp and guff models, qwen VL is not compatible with llama.cpp as it uses a different archetechure because of its vision. it wont be added anytime soon likley.

llama.cpp support qwen-vl now: https://github.com/ggerganov/llama.cpp/pull/10361

**Comment by chigkim** [comment]
Yes, llama.cpp supports qwen2-vl now.

https://huggingface.co/bartowski/Qwen2-VL-72B-Instruct-GGUF

https://huggingface.co/bartowski/Qwen2-VL-7B-Instruct-GGUF

https://huggingface.co/bartowski/Qwen2-VL-2B-Instruct-GGUF

**Comment by akfheaven** [comment]
+1

**Comment by Jovines** [comment]
hope to support

**Comment by feishenebay** [comment]
+1

**Comment by fighter3005** [comment]
I experimented with vllm, but Qwen2 vl on ollama would be just awesome. (with multiple image support).

**Comment by bingbing6** [comment]
Can we use Qwen2-VL now? It's the best visual model, after all

**Comment by Cheesper** [comment]
> Yes, llama.cpp supports qwen2-vl now.
> 
They stupidly droped multimodal from llama-server 9 month ago, so no online VLM, only local

**Comment by billysb** [comment]
This model would be so perfect for the kind of thing I'm working on with tool calls. Support for this model would be very useful.

**Comment by JustLikeThis101** [comment]
Does anyone know if there is currently a team supporting the use of the qwen2-VL model on Ollama?

**Comment by xihuai18** [comment]
https://github.com/ollama/ollama/commit/8c9fb8eb73afc220e8bf99772572096b6498b748
will this commit mean supporting qwen vl

**Comment by bfroemel** [comment]
What's the status here? anyone working on this? ;)

**Comment by Wooonster** [comment]
+1

**Comment by franz101** [comment]
Curios, would work just pulling it from huggingface?

**Comment by MIC-BO** [comment]
support +1

**Comment by pdevine** [comment]
I posted this in #8620, but just so it doesn't get lost:

TLDR; we're working on this, it's going to take a while

Long version: we tried getting this working w/ llama.cpp but it was very hacked together and we couldn't get reasonable quality out of it. We could have shipped a that version but I don't think anyone would have been particularly happy. We are looking to get it working w/ the new engine, but we're still a ways away from getting the new engine out (I could give a timeline for that, but it's invariably going to be wrong).

**Comment by MoRocety** [comment]
cheers!

**Comment by nguyenhongson1902** [comment]
can't wait to see this model w/ ollama

**Comment by tobalo** [comment]
+1 thank you and kudos to all the team. Llama team keep being the western champions of open source AI!

**Comment by EddyChen** [comment]
support +1

**Comment by sxsnwangrenfei** [comment]
+1

**Comment by zytoh0** [comment]
> I posted this in #8620, but just so it doesn't get lost:
> 
> TLDR; we're working on this, it's going to take a while
> 
> Long version: we tried getting this working w/ llama.cpp but it was very hacked together and we couldn't get reasonable quality out of it. We could have shipped that version but I don't think anyone would have been particularly happy. We are looking to get it working w/ the new engine, but we're still a ways away from getting the new engine out (I could give a timeline for that, but it's invariably going to be wrong).

Since you're still sometime away, is it possible that you release this hacked version first for us to use and then update it once you are satisfied? Please 🙏 🙏 🙏 🙏 🙏 thumbs up if anyone agrees.

**Comment by aole** [comment]
+3.14

**Comment by poo0054** [comment]
Can you release a beta version first? I need Ollama's API.

**Comment by seblessa** [comment]
Is there an ETA for this ?

**Comment by misaka-20002** [comment]
any ETA

**Comment by wcool1** [comment]
look foward to

**Comment by sam8890** [comment]
+1

**Comment by 3unnycheung** [comment]
https://github.com/ollama/ollama/issues/8424 hope so

**Comment by pdevine** [comment]
No ETA yet, but I'm currently working on the new image processing code for the new engine, so hopefully soon-ish 🤞 @jmorganca did the original change, and I know everyone wants this model desperately, but please just hang tight for a bit longer. We'd rather publish a working model than publish a half working one.

**Comment by olumolu** [comment]
+1 for me please

**Comment by caiyajunx** [comment]
+1

**Comment by Praveenstein** [comment]
+1

**Comment by wk19921225** [comment]
+1

**Comment by ianmage** [comment]
+1

**Comment by crazyn2** [comment]
+1

**Comment by EuPhobos** [comment]
Stop "+1" please!
Just use "Subscribe" button..

**Comment by xqq965** [comment]
+1

**Comment by xiasi0** [comment]
Is there any news here?

**Comment by poo0054** [comment]
What difficulties are you facing now and how long will it take to get support? Can you share it with us?

**Comment by kevin-pw** [comment]
Awesome! Looks like the Ollama team is [actively working on a Qwen 2.5 VL branch](https://github.com/ollama/ollama/compare/main...brucemacd/qwen25vl).

**Comment by VasiliyMooduckovich** [comment]
i tried to put it myself into ollama but i got this error

`copying file sha256:9c5ae00e602b8860cbd784ba82a8aa14e8feecec692e7076590d014d7b7fdafa 100%
converting model
Error: unsupported architecture "Qwen2_5_VLForConditionalGeneration"
`

**Comment by wuaigleuiogueae** [comment]
+1

**Comment by eroyee** [comment]
It seems to me that unless one is able to contribute in terms of coding, the best thing that could be done in terms of 'support' is let the devs who _are_ contributing get on with it rather than need to respond here?

Practically, keep an eye on the link that @kevin-pw posted, you can see what work is being done there, and whenever it becomes officially available I expect you'd see the model appear near the top of the list here: https://ollama.com/library?sort=newest

Otherwise my thanks to those who have and are working on this project, I very much appreciate the opportunity to learn how these things work and I'm sure Qwen2.5-VL will come along in due course :)

**Comment by ddpasa** [comment]
> It seems to me that unless one is able to contribute in terms of coding, the best thing that could be done in terms of 'support' is let the devs who _are_ contributing get on with it rather than need to respond here?
> 
> Practically, keep an eye on the link that [@kevin-pw](https://github.com/kevin-pw) posted, you can see what work is being done there, and whenever it becomes officially available I expect you'd see the model appear near the top of the list here: https://ollama.com/library?sort=newest
> 
> Otherwise my thanks to those who have and are working on this project, I very much appreciate the opportunity to learn how these things work and I'm sure Qwen2.5-VL will come along in due course :)

The real work to support Qwen2-VL is happening in llama.cpp.  They already have a CLI for supporting it here: https://github.com/ggml-org/llama.cpp/pull/10361

However llama-server does not yet support VLs, so it's a little tricky for ollama to support them.

In the mean time I recommend using llama.cpp's qwen executable as a stopgap.

**Comment by pdevine** [comment]
Hey everyone. Thanks for you patience on this. We are *very* close to getting it out but ran into some snags with the way the vision model's patch embeddings work. Note that this is in the ollama engine, and not based on the llama.cpp change referenced above.

**Comment by barseghyanartur** [comment]
Please, also include Qwen2.5-Omni: https://github.com/QwenLM/Qwen2.5-Omni

**Comment by 3unnycheung** [comment]
> Please, also include Qwen2.5-Omni: https://github.com/QwenLM/Qwen2.5-Omni

https://github.com/ollama/ollama/compare/main...qwen25omni

**Comment by euberdeveloper** [comment]
any progress?

**Comment by eroyee** [comment]
> any progress?

**Yes**.

Why don't you just have a look a few comments up and follow the links - you'll see quite a lot of work on this. eg. [https://github.com/ollama/ollama/compare/main...brucemacd/qwen25vl](url)

**Comment by LFd3v** [comment]
@BruceMacD The model page in ollama.com says it supports video, but I tried with an MP4 and it was not processed? Known but or indeed there is no support for video and it is not planned? Thank you.

**Comment by BruceMacD** [comment]
@LFd3v our implementation doesn't support video at this point, that part of the description was misleading. Sorry about that, I've updated the details on the website.

**Comment by eroyee** [comment]
> Latest ollama doesn't offload this model to gpu despite offloading others like qwen3.

FYI that's **not** my experience (Ubuntu 24.04, Nvidia 1070 and latest drivers), nvidia-smi -l shows healthy use of the GPU with the 7b model...

**Comment by jmtatsch** [comment]
@BruceMacD what is missing for supporting video?

**Comment by BruceMacD** [comment]
@jmtatsch A few things would be needed, Ollama doesn't have the ability to send video from the client the server and the model implementation only processes the one image frame while video implementations process multiple video frames at different steps in the multimodal model.
